edgedb/base_client.py

```diff
@@ -287,6 +287,25 @@ async def _execute(self, execute_context: abstract.ExecuteContext) -> None:
                 ),
             )
 
+    async def describe(
+        self, describe_context: abstract.DescribeContext
+    ) -> abstract.DescribeResult:
+        cardinality, in_dc, out_dc, _ = await self._protocol._parse(
+            describe_context.query,
+            reg=protocol.CodecsRegistry(),
+            inline_typenames=True,
+            inline_typeids=True,
```

Comment:
These are actually not correct - they only add type names and IDs into the `Execute` result not the `Parse` result (an exception is non-builtin scalars), so we'll probably need a new `CompilationFlag` (even though it doesn't affect compilation, but there's no other fields in the `Parse` message to set this, unless we change the protocol) in order to include the type names in the `Parse` result.

Reply:
We'll use introspection query to get all names.